services/tenants/tenants/application_test.go

```diff
@@ -0,0 +1,303 @@
+package tenants
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"sensorbucket.nl/sensorbucket/internal/pagination"
+)
+
+func TestCreateParentTenantDoesNotExist(t *testing.T) {
+	// Arrange
+	store := tenantStoreMock{
+		GetTenantByIdFunc: func(id int64) (Tenant, error) {
+			assert.Equal(t, int64(132), id)
+			return Tenant{}, ErrTenantNotFound
+		},
+	}
+	s := service{
+		tenantStore: &store,
+	}
+
+	// Act
+	parent := int64(132)
+	_, err := s.CreateNewTenant(TenantDTO{
+		ParentID: &parent,
+	})
+
+	// Assert
+	assert.ErrorIs(t, err, ErrParentTenantNotFound)
+	assert.Len(t, store.GetTenantByIdCalls(), 1)
+}
+
+func TestCreateParentTenantCantBeRetrieved(t *testing.T) {
+	// Arrange
+	expErr := fmt.Errorf("some weird database error has occurred")
+	store := tenantStoreMock{
+		GetTenantByIdFunc: func(id int64) (Tenant, error) {
+			assert.Equal(t, int64(675), id)
+			return Tenant{}, expErr
+		},
+	}
+	s := service{
+		tenantStore: &store,
+	}
+
+	// Act
+	parent := int64(675)
+	_, err := s.CreateNewTenant(TenantDTO{
+		ParentID: &parent,
+	})
+
+	// Assert
+	assert.ErrorIs(t, err, expErr)
+	assert.Len(t, store.GetTenantByIdCalls(), 1)
+}
+func TestCreateParentTenantIsNotActive(t *testing.T) {
+	// Arrange
+	store := tenantStoreMock{
+		GetTenantByIdFunc: func(id int64) (Tenant, error) {
+			assert.Equal(t, int64(675), id)
+			return Tenant{
+				ID:    675,
+				State: Archived,
+			}, nil
+		},
+	}
+	s := service{
+		tenantStore: &store,
+	}
+
+	// Act
+	parent := int64(675)
+	_, err := s.CreateNewTenant(TenantDTO{
+		ParentID: &parent,
+	})
+
+	// Assert
+	assert.ErrorIs(t, err, ErrParentTenantNotFound)
```

Comment:
We could choose to create a `ErrTenantNotActive` error and assign http code 404 using `web.NewError(...)`